app/src/js/components/home.js

```diff
@@ -43,14 +46,33 @@ import {
   kibanaGatewayExecutionSuccessesLink,
   kibanaAllLogsLink,
 } from '../utils/kibana';
-
+// import { initialValuesFromLocation } from '../utils/url-helper';
+import Datepicker from './Datepicker/Datepicker';
+import { urlDateFormat, urlDateProps } from '../utils/datepicker';
 import { strings } from './locale';
 
-const { recent, updateInterval } = _config;
+/*
+ * If this is a shared URL, grab the date and time and update the datepicker
+ * state to reflect the values.
+ * @param {Object} props - Home component's input props.
+ */
+const updateDatepickerStateFromQueryParams = (props) => {
+  const { queryParams } = props;
+  const values = {...queryParams};
+  if (!isEmpty(queryParams)) {
+    for (const value in values) {
+      if (urlDateProps.includes(value)) {
+        values[value] = moment.utc(values[value], urlDateFormat).toDate();
+      }
+    }
+    values.dateRange = {value: 'Custom', label: 'Custom'};
+    props.dispatch({type: 'DATEPICKER_DATECHANGE', data: {...props.datepicker, ...values}});
+  }
+};
```

Comment:
Nothing actionable, but we may want to eventually move this outside of the home component if we want to be able to use a shareable date on other pages.

Reply:
Oh yeah good catch! I remembered this at one point and never asked anyone about it.  It does seem like we could move it to the Datepicker.js file.  I will do that in the next PR if it's easy/possible.

Reply:
I just added this to my next PR that also has the speed improvements.